tests/CMakeLists.txt

```
@@ -3,16 +3,45 @@ set(PHPUNIT "${CDash_SOURCE_DIR}/vendor/bin/phpunit")
 
 function(add_vue_test TestName)
     add_test(
-        NAME ${TestName}
-        COMMAND "node_modules/.bin/jest" "tests/${TestName}.spec.js"
-        WORKING_DIRECTORY "${CDash_SOURCE_DIR}"
-        )
+      NAME ${TestName}
+      COMMAND "node_modules/.bin/jest" "tests/${TestName}.spec.js"
+      WORKING_DIRECTORY "${CDash_SOURCE_DIR}"
+    )
 endfunction(add_vue_test)
 
+function(add_cypress_test TestName)
+  add_test(
+    NAME cypress/${TestName}
+    COMMAND ${NPX_EXE} cypress run --project ${CDash_SOURCE_DIR} --spec ${CDash_SOURCE_DIR}/tests/cypress/e2e/${TestName}.cy.js
+  )
+  set_tests_properties(
+      cypress/${TestName}
+    PROPERTIES
+      ENVIRONMENT "HOME=/cdash;" # Cypress tries to put stuff in our home directory, which doesn't work for /var/www.
+  )
+endfunction(add_cypress_test TestName)
```

Even in older versions of CMake (< 3.14 I believe), the argument to [`endfunction`](https://cmake.org/cmake/help/latest/command/endfunction.html) should be just the name of the function, without the names of the rest of the arguments. In newer version of cmake, there's no need to pass any arguments to this function: 
```suggestion
endfunction()
```
If for now we wish to follow the syntax used across the other CMake files in the repo, we should at least remove the extra argument:
```suggestion
endfunction(add_cypress_test)
```